pkg/vm/engine/disttae/logtailreplay/partition_state.go

```diff
@@ -660,3 +666,60 @@ func (p *PartitionState) consumeCheckpoints(
 	p.checkpoints = p.checkpoints[:0]
 	return nil
 }
+
+func (p *PartitionState) truncate(ids [2]uint64, ts types.TS) {
```

Comment:
This method should be moved to Partition. And never modify a PartitionState, copy and modify instead. See Partition.ConsumeCheckpoints and Partition.Mutate for how to do modifications.